cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -369,6 +369,29 @@ private void setShaclDatatype(Configuration configuration,
 
     Resource datatypeValue = ResourceFactory.createResource(datatypes.get(0));
     shacl.add(propertyShape, Shacl.datatype, datatypeValue);
+
+    if (RDF.langString.equals(datatypeValue)) {
+      setLanguageIn(rdfStoreService, shacl, targetClass, path, propertyShape);
+    }
+  }
+
+  private void setLanguageIn(@Nonnull RdfStoreService rdfStoreService,
+                             @Nonnull Model shacl,
+                             @Nonnull Resource targetClass,
+                             @Nonnull Resource path,
+                             @Nonnull Resource propertyShape) {
+
+    List<Map<String, RDFNode>> rows = getRowsForTypeAndProperty(rdfStoreService, "select-languages.sparql.spel",
+                                                                targetClass, path);
+    if (rows.isEmpty()) return;
+
+    List<String> languages = rows.stream()
+                                 .map(row -> row.get(0).asLiteral().getString())
```

Comment:
row.get(0)
is this ok? should be row.get("varName")
or use iterator on key

also add .distinct() or use set in case we have 2 literals for same language

Reply:
good catch. I now added a function in PaginatedQuery 
```
  public <T> List<T> convertSingleColumnToList(@Nonnull List<Map<String, RDFNode>> results,
                                               @Nonnull Function<RDFNode, T> conversion)
```